### PR TITLE
Fix Faktory::Job.set ignores faktory_options

### LINKED ISSUE #62: Faktory::Job.set ignores class-defined faktory_options
I'm not sure if this is intended behavior, but I was surprised by it.

In [Getting Started](https://github.com/contribsys/faktory_worker_ruby/wiki/Getting-Started#faktoryjob), there's a code comment that suggests that use of `Faktory::Job.set` "dynamically override[s]" options. But, while it does, it also drops any other options that are defined on the Job. So, to use `set` you have to pass all the options to it, in order to override, say, just 1.

Here's some code that demonstrates the behavior, using the example job in the docs:
```ruby
class SomeJob
  include Faktory::Job
  faktory_options queue: "critical", custom: { unique_for: 10.minutes.to_i }
  def perform(*)
  end
end

it 'fails to use the critical queue' do
  SomeJob.set(custom: { unique_for: nil }).perform_async(1)
  assert_equal 1, Faktory::Queues['default'].size
  assert_equal 0, Faktory::Queues['critical'].size
end
```

### PR DESCRIPTION
**Pull request by ahacop**
Fixes #62.

### CODE REVIEW

**Comment by mperham** [comment]
Does `Job.set(queue: "foo").set(queue: "bar").perform_async` work as expected? Seems like Setter needs to implement `set` too.

**Comment by ahacop** [comment]
> Does `Job.set(queue: "foo").set(queue: "bar").perform_async` work as expected? Seems like Setter needs to implement `set` too.

Didn't realize that was a thing. How's this?

File: lib/faktory/job.rb [hunk 1]
```diff
@@ -56,6 +56,10 @@ def initialize(opts)
         @opts = opts
       end
 
+      def set(opts)
+        self.class.new(@opts.merge(opts))
```
Comment (mperham) [inline lib/faktory/job.rb:60]: Would it be reasonable to merge @opts directly rather than allocating a new instance?

Comment (ahacop) [reply lib/faktory/job.rb]: Yup. Fixed in 1b532f2.

**Comment by mperham** [comment]
This looks wonderful but I think you rightfully point out that the lack of deep merge on the custom attribute is a footgun that will hurt someone eventually. I think we need to have it.

**Comment by ahacop** [comment]
Yeah, I agree. I can write something.

**Comment by ahacop** [comment]
Wasn't really sure where to put the deep_merge method, since the Util module seemed more Faktory related.

Also, while writing the deep_merge, discovered that the actual merge in the prior commits was happening in the stringification of the keys in client_push as in `{ foo: 1, 'foo' => 2}.stringify # => { 'foo' => 2 }`

**Comment by ahacop** [comment]
Thanks!

**Comment by mperham** [comment]
Thank you for not only opening the issue but contributing the fix!

**Pull request merged**